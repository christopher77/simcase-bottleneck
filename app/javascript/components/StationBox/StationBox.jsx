import React from "react";
import Flag from "../ui/Flag/Flag.jsx";
import styles from "./StationBox.module.css";
import classNames from "classnames";

function StationBox(props) {
  const {
    numStation,
    station,
    machinesAvailable,
    round,
    feedback = false,
    flagDirection,
    expectedMachines,
    buttonRef
  } = props;

  const addMachines = station => {
    if (machinesAvailable > 0) {
      props.addMachinesToStation(station, round);
      props.decrementMachinesAvailable(round);
    }
    buttonRef.current.focus();
  };

  const isCorrect =
    station.machinesAssigned ===
    Object.values(expectedMachines)[numStation - 1];
  const classStyle = `buttonContainer${numStation}`;

  return (
    <div className={styles[classStyle]}>
      <Flag
        isCorrect={!feedback || isCorrect}
        solution={station.solution}
        numStation={numStation}
        flagDirection={flagDirection === "left" ? "left" : "right"}
      />
      <button
        className={classNames([styles.button], {
          [styles.buttonCorrect]: feedback && isCorrect,
          [styles.buttonFail]: feedback && !isCorrect
        })}
        onClick={() => addMachines(numStation)}
        disabled={feedback}
      >
        {station.machinesAssigned}
        {!isCorrect && (
          <sup>
            {feedback && Object.values(expectedMachines)[numStation - 1]}
          </sup>
        )}
      </button>
      <h5
        className={classNames([styles.capitalLetter], {
          [styles.capitalLetter]: feedback && isCorrect,
          [styles.capitalWrong]: feedback && !isCorrect
        })}
      >
        {station.stationName}
      </h5>
    </div>
  );
}

export default StationBox;
