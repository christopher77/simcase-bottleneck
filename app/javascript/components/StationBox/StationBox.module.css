.buttonContainer1 {
  position: absolute;
  left: 29px;
  top: 72px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.buttonContainer2 {
  position: absolute;
  left: 117px;
  top: 120px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.buttonContainer3 {
  position: absolute;
  left: 250px;
  top: 169px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.buttonContainer4 {
  position: absolute;
  left: 168px;
  top: 240px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.buttonContainer5 {
  position: absolute;
  left: 46px;
  top: 284px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.button {
  width: 45px;
  height: 45px;
  border-radius: 2px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  font-weight: 400;
  background: linear-gradient(
    180deg,
    #ffffff 0%,
    #ffffff 0.01%,
    #ffffff 47.4%,
    #ececec 100%
  );
  box-shadow: 0px 0px 4px 2px rgba(115, 192, 86, 0.8);
  border-radius: 5px;
  border-color: transparent;
}

.buttonCorrect {
  composes: button;
  animation: scaleCenter 0.4s cubic-bezier(0.39, 0.575, 0.565, 1) both;
}

.buttonFail {
  composes: button;
  box-shadow: 0px 0px 4px 2px rgba(201, 20, 20, 0.8);
  border-radius: 5px;
  animation: shakeHorizontal 0.4s cubic-bezier(0.455, 0.03, 0.515, 0.955) 0.4s
    both;
  color: black;
}

.button sup {
  vertical-align: super;
  font-size: small;
  position: relative;
  left: 4px;
  bottom: 8px;
  color: #ff7070;
}

.capitalLetter:first-letter {
  color: #73c056;
  font-weight: bold;
}

.capitalWrong:first-letter {
  color: #ff7070;
  font-weight: bold;
}

.capitalLetter,
.capitalWrong {
  font-weight: 400;
}

@keyframes scaleCenter {
  0% {
    transform: scale(0.5);
  }
  50% {
    transform: scale(1.5);
  }
  100% {
    transform: scale(1);
  }
}

@keyframes shakeHorizontal {
  0%,
  100% {
    transform: translateX(0);
  }
  10%,
  30%,
  50%,
  70% {
    transform: translateX(-4px);
  }
  20%,
  40%,
  60% {
    transform: translateX(4px);
  }
  80% {
    transform: translateX(1px);
  }
  90% {
    transform: translateX(-1px);
  }
}
