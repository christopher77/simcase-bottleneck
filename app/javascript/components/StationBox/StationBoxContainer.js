import { connect } from "react-redux";
import StationBox from "./StationBox";
import {
  addMachinesToStation,
  decrementMachinesAvailable
} from "../../redux/actions/RoundsActions";

const mapStateToProps = (state, ownProps) => {
  return {
    station: ownProps.station,
    numStation: ownProps.numStation,
    machinesAvailable: ownProps.machinesAvailable,
    round: ownProps.round,
    isFeedback: ownProps.isFeedback,
    expectedMachines: ownProps.expectedMachines
  };
};

const mapDispatchToProps = {
  addMachinesToStation: addMachinesToStation,
  decrementMachinesAvailable: decrementMachinesAvailable
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StationBox);
