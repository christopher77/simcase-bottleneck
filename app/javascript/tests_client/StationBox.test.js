import React from "react";
import { render } from "@testing-library/react";
import StationBox from "../components/StationBox/StationBox";

test("StationBox component", () => {
  const { asFragment } = render(
    <StationBox
      expectedMachines={{}}
      numStation={{}}
      station={{}}
      machinesAvailable={{}}
      round={{}}
      feedback={false}
      flagDirection={{}}
    />
  );
  expect(asFragment).toMatchSnapshot();
});
