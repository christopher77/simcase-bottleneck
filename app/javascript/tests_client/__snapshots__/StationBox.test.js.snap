// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationBox component 1`] = `[Function]`;
